src/sdk/cpp/lib/include/context.hpp

```diff
@@ -282,13 +282,14 @@ class CallContext : public EventContext {
     void GatherAndReply(gather_and_reply_func_ptr func, timeout_func_ptr timeout_func, Args... args) {
         auto now_ms = GetEpochTimeMs();
         auto tokens = std::vector<CallToken>{args...};
-        auto prepared_reply = PreparedFunctionReply{talent_id_, feature_, subject_, channel_, call_, return_topic_, publisher_};
+        auto prepared_reply = PreparedFunctionReply{talent_id_, feature_, event_, return_topic_, publisher_};
```

Comment:
Event carries more then these 3 attributes. Will this lead to more overhead?

Reply:
There's small overhead but not providing the event as an argument will add an additional five arguments to the `PreparedFunctionReply` constructor. I'll change it if you insist but maybe it's better to leave it as is.